community/kernel/src/main/java/org/neo4j/kernel/impl/store/CommonAbstractStore.java

```diff
@@ -545,9 +520,10 @@ public long nextId()
      */
     public void freeId( long id )
     {
-        if ( idGenerator != null )
+        IdGenerator generator = this.idGenerator;
```

Comment:
I think `idGenerator` field is only assigned during startup and is never modified concurrently with transaction threads. Isn't it true?